platform/jobs/RetirementJobs.groovy

```diff
@@ -374,6 +374,11 @@ job('retirement-partner-reporter') {
 
     // Only one of these jobs should be running at a time per environment
     concurrentBuild(true)
+    throttleConcurrentBuilds {
+        // A maxTotal of 0 implies unlimited simultaneous jobs, but below we
+        // restrict one build per environment.
+        maxTotal(0)
+    }
     configure { project ->
```

Comment:
This change was unrelated but I tossed it into this PR anyway.  It just adds a missing section to actually enable throttling to prevent multiple concurrent builds on the same environment.